### PR TITLE
Add description to environment variable model

### PR DESCRIPTION
**Pull request by haleystorm**
🚧 WIP

Done so far...
Initial controller and view modifications to set and view environment variable descriptions

Left to do...
Add unit tests
Add screenshots

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link:

### Risks
- Level: Med - new feature for environment_variables

### CODE REVIEW

File: plugins/env/app/controllers/environment_variables_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 class EnvironmentVariablesController < ApplicationController
-  before_action :authorize_admin!, except: [:index]
+  before_action :authorize_admin!, except: [:index, :show, :update]
+  before_action :environment_variable, only: [:show]
```
Comment (grosser) [inline plugins/env/app/controllers/environment_variables_controller.rb]: `find_environment` would be more consistent with other controllers, should also use it for update (and edit if that exists)

File: plugins/env/app/controllers/environment_variables_controller.rb [hunk 2]
```diff
@@ -10,8 +11,29 @@ def index
     @pagy, @environment_variables = pagy(scope, page: params[:page], items: 30)
   end
 
+  def show
+    render 'form'
```
Comment (grosser) [inline plugins/env/app/controllers/environment_variables_controller.rb]: name the view show and remove the `def show`

File: plugins/env/app/controllers/environment_variables_controller.rb [hunk 3]
```diff
@@ -10,8 +11,29 @@ def index
...
+  end
+
+  def update
+    environment_variable.attributes = attributes
+    environment_variable.save!
```
Comment (grosser) [inline plugins/env/app/controllers/environment_variables_controller.rb]: can combine these 2 lines to `environment_variable.update_attributes!(attributes)`

File: plugins/env/app/controllers/environment_variables_controller.rb [hunk 4]
```diff
@@ -10,8 +11,29 @@ def index
...
+    redirect_to action: :show
```
Comment (grosser) [inline plugins/env/app/controllers/environment_variables_controller.rb]: `redirect_to environment_variable ` should also work

File: plugins/env/app/controllers/environment_variables_controller.rb [hunk 5]
```diff
@@ -10,8 +11,29 @@ def index
...
   def destroy
     EnvironmentVariable.find(params.require(:id)).destroy!
     head :ok
   end
+
+  def environment_variable
+    @variable ||= EnvironmentVariable.find(params[:id])
+  end
+
+  def attributes
```
Comment (grosser) [inline plugins/env/app/controllers/environment_variables_controller.rb]: `environment_attributes` would be more consistent with other controllers

File: plugins/env/app/views/environment_variables/form.html.erb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  disabled = can_edit ? '' : 'disabled'
+  crumbs = [["Environment variables", EnvironmentVariableGroup]]
+%>
+
+<%= manual_breadcrumb(crumbs + [[title, nil]]) %>
+
+<%= page_title title %>
+
+<section>
+  <% if !can_edit %>
```
Comment (grosser) [inline plugins/env/app/views/environment_variables/form.html.erb]: `if !` -> `unless`

File: plugins/env/app/views/environment_variables/form.html.erb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      You must either be an admin or have project admin privileges for all projects that use this environment variable to
+      edit it.
+    </div>
+  <% end %>
+  <%= form_for @variable, html: { class: "form-horizontal" } do |form| %>
+    <fieldset <%= disabled %>>
+      <%= form.input :name %>
+      <%= form.input :description, as: :text_area, input_html: {size: "80x4"}%>
+    </fieldset>
+    <%= form.actions %>
```
Comment (grosser) [inline plugins/env/app/views/environment_variables/form.html.erb]: should not show the button if user cannot click it

File: plugins/env/app/views/environment_variables/form.html.erb [hunk 8]
```diff
@@ -0,0 +1,26 @@
+<%
+  title = "Edit #{@variable.name}"
+  can_edit = can?(:write, :environment_variable_groups, @variable)
+  disabled = can_edit ? '' : 'disabled'
```
Comment (grosser) [inline plugins/env/app/views/environment_variables/form.html.erb]: can do that inline

**Comment by grosser** [comment]
maybe add an extra field to the row instead that expends with js ?
`.on('focus')` ... would need no new UI/flow ...

**Comment by haleystorm** [comment]
Closing - re-prioritized

**Pull request closed without merging**